### PR TITLE
Fixes handling of relative URLs in the post body

### LINKED ISSUE #148: Review handling of URLs in the post body
I'm receiving an "INVALID_DOCUMENT" error when publishing various posts. Attached is the JSON. I was unable to see anything specifically wrong with the document.

[article-2692.zip](https://github.com/alleyinteractive/apple-news/files/214080/article-2692.zip)

### PR DESCRIPTION
**Pull request by bcampeau**
Also removes anchors, which are not valid in Apple News.

Also adds corresponding unit tests.

Closes #148

### CODE REVIEW

File: includes/apple-exporter/class-parser.php [hunk 1]
```diff
@@ -129,4 +134,56 @@ private function _parse_markdown( $html ) {
...
+	 * both HTML and Markdown format.
+	 *
+	 * @access private
+	 * @param string $html
+	 * @return string The clean HTML
+	 */
+	private function _clean_html( $html ) {
+		// Match all <a> tags via regex.
+		// We can't use DOMDocument here because some tags will be removed entirely.
+		preg_match_all( '/<a(?:.*?)(href|name)=(?:"|\')(.*?)(?:"|\')(?:.*?)>(.*?)<\/a>/m', $html, $a_tags );
```
Comment (mslinnea) [inline includes/apple-exporter/class-parser.php]: Do you need that m modifier here? Isn't that only applicable if you're using `^` or `$` in the regex?

File: tests/apple-exporter/test-class-parser.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<?php
+
+use Apple_Exporter\Parser as Parser;
+
+class Parser_Test extends WP_UnitTestCase {
+
+	public function testCleanInvalidLinks() {
```
Comment (mslinnea) [inline tests/apple-exporter/test-class-parser.php]: empty function

**Comment by mslinnea** [comment]
🐴

File: includes/apple-exporter/class-parser.php (hunk 1 above)
Comment (bcampeau) [inline includes/apple-exporter/class-parser.php]: I believe it's still required for multi-line matches

**Pull request merged**